src/live-programming.rst

```diff
@@ -36,8 +36,132 @@ Direct Manipulation Programming
+In this compendium, our scope covers only a tiny amount of the whole problem space (i.e. **tools** and **languages** for game development).
+
+Professional Platforms for Game Development
+-------------------------------------------------
+In recent years, game engines such as Unity3D and Unreal Engine have been the most popular platforms for game development.
+A game engine simplifies the task of the programmer by offering convenient abstractions for the hardware and operating systems on top of which the game runs.
+Game engines also provides reusable components such as physics, game inputs, rendering, and Artificial Intelligence (AI).
+Entity-component system and event-handling mechanism have been the major architecture of game engines nowadays.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'anderson2008case'
+
+  This article presents a number of potential research topics with regard to the game engine development and architecture design.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'messaoudi2015dissecting'
+
+  This paper dissected and analyzed the CPU and GPU performances of one of the most popular game engines, namely Unity3D.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'elvezio2018mercury'
+
+  This paper presented a framework as an improvement for the existing game engine architecture.
+  In 3D interactive systems that are developed using game engines, User Interface (UI) components are organized in a hierarchy that is used to propagate events among vertically connected components.
+  However, programmers have to connect horizontal components manually and register/unregister events as needed in order to enable communications between those horizontal components.
+  This paper introduced a messaging framework, Mercury, to facilitate communication among components.
+  This framework simplifies message propagation for inter-component communication and provides support for UI management and sharing.
+
+Non-imperative Programming Languages for Game Development
+---------------------------------------------------------
+Most game development platforms require highly domain-specific knowledge of imperative programming languages, which has a steep learning curve for end-users.
+For example, Unity3D employs C# as its scripting language and Unreal Engine uses C++ as its scripting language.
+In this subsection we examine Functional Reactive Programming as a declarative programming language in contrast to imperative programming languages.
+We also introduce visual block-based programming languages and visual dataflow programming languages for game development.
```

Comment:
"Functional Reactive Programming as a declarative programming language in contrast to imperative programming languages.  We also introduce visual block-based programming languages and visual dataflow programming languages for game development. " -> "functional reactive programming, visual block-based languages, and visual dataflow languages for game development."